### DISCUSSION TITLE
Appium 2.0 with appium flutter driver capabilities for IOS

### DISCUSSION THREAD

**Discussion by Harsha-Gelato** [appium/appium#17273] [General]
Hi All, 

I have recently started with automation of my flutter application using Appium 2.0 with appium flutter driver. For **android** it is **working correctly**. For **IOS**, I am facing error. I am using below capabilities:

```
// Appium capabilities
config.capabilities = [
    {
        'appium:platformName': 'iOS',
        'appium:noReset': false,
        'appium:fullReset': false,
        'appium:maxInstances': 1,
        **'appium:automationName': 'Flutter'**,
        'appium:deviceName': 'iPhone 13',
        'appium:platformVersion': '15.4',
        'appium:app': '../build/ios/iphonesimulator/Runner.app',
        'appium:appWaitDuration': 20000,
        'appium:newCommandTimeout': 90000,
        'appium:webviewConnectTimeout': 20000,
        'appium:launchTimeout': 5000,
        'appium:fullContextList': true,
        'appium:autoAcceptAlerts': true,
        'appium:chromedriverExecutable': '',
        'appium:chromeOptions': {
            'w3c': false,
            'args': [
                '--disable-popup-blocking',
            ],
          },
    },
];
```
__When I replaced **'appium:automationName': 'Flutter'**, with **'appium:automationName': 'XCUITest'**, it works perfectly._

Can someone please tell me what I am missing here?
[appium_logs.txt](https://github.com/appium/appium/files/9197545/appium_logs.txt)

<img width="1192" alt="Screenshot 2022-07-27 at 11 37 46" src="https://user-images.githubusercontent.com/103107117/181227444-a2ef1b2b-249e-48d4-a529-2857af297029.png">

**Comment by RSM-SBA** [comment]
i would say this is the cause
[FlutterDriver] "ext.flutter.driver" is not found in "extensionRPCs" 

**from appium flutter driver page :**
Also, ensure that your Flutter AUT has enableFlutterDriverExtension() before runApp. Then, please make sure your app imported [flutter_driver](https://api.flutter.dev/flutter/flutter_driver/flutter_driver-library.html) package as well.

if you check the issue page of the flutter driver repo, you can check some similar issues if those solve your problem

**Reply by Harsha-Gelato** [reply]
I have enabled it, that is the reason my below android capabilities are working
```
// Appium capabilities
config.capabilities = [{
    'appium:platformName': 'Android',
    'appium:deviceName': 'Pixel 5 API 30 2',
    'appium:automationName': 'Flutter',
    'appium:app': '/Users/harsha.sharmagelato.com/Dev/work/repos/gelato-api-mobile/build/app/outputs/flutter-apk/app-stage-debug.apk',
    'appium:noReset': false, // on Android will keep your app cache saved and with new Appium session there will be no need to log in again.
    'appium:fullReset': false,
    'appium:maxInstances': 1,
    'appium:platformVersion': '11',
    'appium:appWaitDuration': 20000,
    'appium:newCommandTimeout': 12000,
    'appium:webviewConnectTimeout': 20000,
    'appium:launchTimeout': 5000,
    'appium:fullContextList': false,
    'appium:autoAcceptAlerts': true,
    'appium:chromedriverExecutable': './node_modules/chromedriver/lib/chromedriver/chromedriver',
    'appium:chromeOptions': {
        'w3c': false,
        'args': [
            '--no-sandbox',
        ],
    },
}];
```
Not sure why it is only happening with IOS.

**Reply by RSM-SBA** [reply]
sorry i have no flutter programming expierience.
But maybe you try a flutter demo app, If this one is running with appium, you can maybe find the mistake.

**Reply by Harsha-Gelato** [reply]
I did it, after following this https://testingbot.com/resources/articles/appium-flutter-app-testing. Thankyou. It is resolved now :)

**Reply by KazuCocoa** [reply]
as https://github.com/appium-userland/appium-flutter-driver#usage

> Your Flutter app-under-test (AUT) must be compiled in debug or profile mode, because Flutter Driver does not support running in release mode.

especially iOS https://github.com/appium-userland/appium-flutter-driver/issues/115, it needs a profile build. Previously it worked with debug build though.

**Comment by changfeifan** [comment]
same

**Comment by RanaToqeer** [comment]
@Harsha-Gelato  can you please help me I am stuck at this error while running Appium flutter driver Android Automation 

Below are my capabilities:


 
  capabilities.setCapability ("appium:platformName", "Android");
        capabilities.setCapability ("appium:deviceName", "emulator-5554");
        capabilities.setCapability ("appium:automationName", "Flutter");
        capabilities.setCapability ("appium:app", "/Users/ranatoqeer/Documents/app-qa-debug.apk");
        capabilities.setCapability ("appium:noReset", false);
        capabilities.setCapability ("appium:fullReset", false);
        capabilities.setCapability("appium:appPackage", "com.islamicfinder.alif.qa");
        capabilities.setCapability("appium:appActivity","com.islamicfinder.alif.MainActivity");
        capabilities.setCapability ("appium:maxInstances", 1);
        capabilities.setCapability ("appium:platformVersion", "14");
        capabilities.setCapability ("appium:appWaitDuration", 20000);
        capabilities.setCapability ("appium:newCommandTimeout", 12000);
        capabilities.setCapability ("appium:webviewConnectTimeout", 20000);
        capabilities.setCapability ("appium:launchTimeout", 5000);
        capabilities.setCapability ("appium:fullContextList", false);
        capabilities.setCapability ("appium:autoAcceptAlerts", true);


But When Application launch in Device then test cases not being started and Application crashed 
And Appium 2.0 Showing the below error 
![image](https://github.com/appium/appium/assets/72913664/37dfe8e1-fb32-42e4-a307-ac8c1a2d35ba)
